slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -866,7 +913,7 @@ void SLAMServiceImpl::ProcessDataAndStartSavingMaps(double data_start_time) {
         map_builder.map_builder_->FinishTrajectory(trajectory_id);
     }
     if (!use_live_data) {
-        BackupLatestMap();
+        if (action_mode != ActionMode::LOCALIZING) BackupLatestMap();
         {
             std::unique_lock<std::shared_mutex> optimization_lock(
```

Comment:
Do we need to optimize the map at all in localization mode?

Reply:
I would say yes because the optimization step can help get cartographer to localize on the map. In the map/dataset I have posted above, the jpeg map/pose fixes itself(i.e. looks aligned) after the optimization step occurs.

Reply:
thanks, can you add that as a comment?

Reply:
Good question, I wonder as well if we should have this instead:

```cpp
    if (action_mode != ActionMode::LOCALIZING && !use_live_data) {
        BackupLatestMap();
```

Reply:
EDIT: Just saw the discussion that happened 40 min ago, so my comment is a outdated.